src/transformers/pipelines/__init__.py

```diff
@@ -696,14 +707,36 @@ def pipeline(
     if isinstance(model, Path):
         model = str(model)
 
+    if is_peft_available() and isinstance(model, str):
+        try:
+            list_remote_files = list_repo_files(model, revision=revision, use_auth_token=use_auth_token)
+        except EntryNotFoundError:
+            if not os.path.isdir(model):
+                raise ValueError(f"Local file or directory {model} does not exist.")
+            list_remote_files = os.listdir(model)
+
+        has_adapter_config = "adapter_config.json" in list_remote_files
+    else:
+        has_adapter_config = False
```

Comment:
Here it would be more simple and more efficient to leverage `cached_file` and directly try to download/load from cache `adapter_config.json`. `list_repo_files` won't work in offline mode or if the Hub is down while `cached_file` will (if the file has been previously tried to be downloaded while online).

Reply:
Makes sense, thanks for flagging this!